library/core/src/str/mod.rs

```diff
@@ -324,7 +323,7 @@ impl str {
     #[allow(unused_attributes)]
     pub const fn as_bytes(&self) -> &[u8] {
         // SAFETY: const sound because we transmute two types with the same layout
-        unsafe { mem::transmute(self) }
+        unsafe { &*(self as *const str as *const [u8]) }
```

Comment:
This comment should probably be updated to "`str` and `[u8]` have the same layout"